aggregator/handler/router/with_ack.go

```diff
@@ -60,10 +61,27 @@ func (d message) Bytes() []byte {
 	return d.buffer.Buffer().Bytes()
 }
 
-func (d message) Size() uint32 {
-	return uint32(len(d.Bytes()))
+func (d message) Size() int {
+	// Use the cap of the underlying byte slice in the buffer instead of
+	// the length of the byte encoded.
+	return d.buffer.Buffer().Buffer().Cap()
 }
 
 func (d message) Finalize(producer.FinalizeReason) {
 	d.buffer.DecRef()
 }
+
+type shardFilter struct {
```

Comment:
Can you move this and the ctor etc to a `shard.go` under `aggregator/handler/filter` package?